sonarts-core/src/rules/noIgnoredInitialValueRule.ts

```diff
@@ -0,0 +1,108 @@
+    ruleName: "no-ignored-initial-value",
+    description: "Function parameters, caught exceptions and foreach variables' initial values should not be ignored",
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-1226",
+    type: "functionality",
+    typescriptOnly: false,
+  };
+
+  public static formatMessage(deadIdentifier: ts.Identifier) {
+    return `Introduce a new variable or use its initial value before reassigning "${deadIdentifier.getText()}".`;
+  }
+
+  public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): tslint.RuleFailure[] {
+    const symbols = SymbolTableBuilder.build(sourceFile, program);
+    return this.applyWithWalker(new Walker(sourceFile, this.getOptions(), program, symbols));
+  }
+}
+
+class Walker extends tslint.ProgramAwareRuleWalker {
+  private lva: LiveVariableAnalyzer;
+
+  constructor(
+    sourceFile: ts.SourceFile,
+    options: tslint.IOptions,
+    program: ts.Program,
+    private readonly symbols: SymbolTable,
+  ) {
+    super(sourceFile, options, program);
+    this.lva = new LiveVariableAnalyzer(this.symbols);
+  }
+
+  protected visitNode(node: ts.Node): void {
+    if (is(node, ...FUNCTION_LIKE)) {
+      const functionLike = node as ts.FunctionLikeDeclaration;
+      const lvaReturn = this.lva.analyzeFunction(functionLike);
+      this.visit(lvaReturn, functionLike.body!, ...functionLike.parameters);
+    
+    } else if (is(node, ts.SyntaxKind.CatchClause)) {
+      const catchClause = node as ts.CatchClause;
+      const lvaReturn = this.lva.analyze(catchClause.block, catchClause.block.statements);
+      this.visit(lvaReturn, catchClause.block, catchClause.variableDeclaration);
+    
+    } else if (is(node, ts.SyntaxKind.ForInStatement, ts.SyntaxKind.ForOfStatement)) {
+      const iterationStatement = node as ts.IterationStatement;
+      const lvaReturn = this.lva.analyze(iterationStatement, [iterationStatement.statement]);      
+      this.visit(lvaReturn, iterationStatement.statement, (iterationStatement as ts.ForInStatement).initializer);
+    }
+ 
+    super.visitNode(node);
+  }
+  
+  private visit(lvaReturn: LVAReturn, root: ts.Node, ... nodes: ts.Node[]) {
+    const {cfg, blockAvailableReads} = lvaReturn;
+    if (cfg && blockAvailableReads) {
+      const symbolsLiveAtFunctionStart = blockAvailableReads.get(cfg.start)!;
+      
+      nodes.forEach(parameter => {
+        descendants(parameter)
+        .filter(descendant => is(descendant, ts.SyntaxKind.Identifier))
+        .forEach(descendant => {
+          const identifier = descendant as ts.Identifier;
+          const symbol = this.getTypeChecker().getSymbolAtLocation(identifier);
+          if (symbol && (symbol.flags == ts.SymbolFlags.BlockScopedVariable || symbol.flags == ts.SymbolFlags.FunctionScopedVariable) && this.isNotUsedElseWhere(symbol, root) && !symbolsLiveAtFunctionStart.has(symbol) && this.symbols.allUsages(symbol).length > 1) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Exceeds maximum line length of 180 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS103)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') == should be === [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1440)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') == should be === [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1440)